msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/AbstractMsalBrokerTest.java

```diff
@@ -44,20 +42,14 @@
 
     protected ITestBroker mBroker = getBroker();
 
-    @Rule(order = 5)
-    public final TestRule brokerSupportRule = new BrokerSupportRule(mBroker);
-
-    @Rule(order = 6)
-    public final TestRule installBrokerRule = new InstallBrokerTestRule(mBroker);
-
-    @Rule(order = 7)
-    public final TestRule powerLiftIncidentRule = new PowerLiftIncidentRule(mBroker);
-
-    @Rule(order = 8)
-    public final TestRule deviceEnrollmentFailureRecoveryRule = new DeviceEnrollmentFailureRecoveryRule();
-
+    @NonNull
     @Override
     public ITestBroker getBroker() {
+        // only initialize once....so calling getBroker from anywhere returns the same instance
```

Comment:
Are these parallelized? (Or could they be in the future?) Does this need to be a `synchronized` call?

Reply:
I can make it synchronized....but this is currently only called once per test case (each test case would create new instance of this class)

Reply:
sorry that wasn't entirely correct

Reply:
It is not parallelized right now......I can make it synchronized....but each test case would create new instance of this class...so it shouldn't really matter